conversion/conversion.go

```diff
@@ -0,0 +1,789 @@
+		p.MaybeReport(atomic.LoadInt64(&rows))
+		return nil
+	}
+	writer := spanner.NewBatchWriter(config)
+	conv.SetDataMode() // Process data in dump; schema is unchanged.
+	conv.SetDataSink(
+		func(table string, cols []string, vals []interface{}) {
+			writer.AddRow(table, cols, vals)
+		})
+	ProcessDump(driver, conv, r)
+	writer.Flush()
+	p.Done()
+
+	return writer, nil
+}
+
+func Report(driver string, badWrites map[string]int64, BytesRead int64, banner string, conv *internal.Conv, reportFileName string, out *os.File) {
+	f, err := os.Create(reportFileName)
+	if err != nil {
+		fmt.Fprintf(out, "Can't write out report file %s: %v\n", reportFileName, err)
+		fmt.Fprintf(out, "Writing report to stdout\n")
+		f = out
+	} else {
+		defer f.Close()
+	}
+	w := bufio.NewWriter(f)
+	w.WriteString(banner)
+
+	summary := internal.GenerateReport(driver, conv, w, badWrites)
+	w.Flush()
+	var isDump bool
+	if strings.Contains(driver, "dump") {
+		isDump = true
+	}
+	if isDump {
+		fmt.Fprintf(out, "Processed %d bytes of %s data (%d statements, %d rows of data, %d errors, %d unexpected conditions).\n",
+			BytesRead, driver, conv.Statements(), conv.Rows(), conv.StatementErrors(), conv.Unexpecteds())
+	} else {
+		fmt.Fprintf(out, "Processed source database via %s driver (%d rows of data, %d unexpected conditions).\n",
+			driver, conv.Rows(), conv.Unexpecteds())
+	}
+	// We've already written summary to f (as part of GenerateReport).
+	// In the case where f is stdout, don't write a duplicate copy.
+	if f != out {
+		fmt.Fprint(out, summary)
+		fmt.Fprintf(out, "See file '%s' for details of the schema and data conversions.\n", reportFileName)
+	}
+}
+
+// getSeekable returns a seekable file (with same content as f) and the size of the content (in bytes).
+func getSeekable(f *os.File) (*os.File, int64, error) {
+	_, err := f.Seek(0, 0)
+	if err == nil { // Stdin is seekable, let's just use that. This happens when you run 'cmd < file'.
+		n, err := getSize(f)
+		return f, n, err
+	}
+	internal.VerbosePrintln("Creating a tmp file with a copy of stdin because stdin is not seekable.")
+
+	// Create file in os.TempDir. Its not clear this is a good idea e.g. if the
+	// pg_dump/mysqldump output is large (tens of GBs) and os.TempDir points to a directory
+	// (such as /tmp) that's configured with a small amount of disk space.
+	// To workaround such limits on Unix, set $TMPDIR to a directory with lots
+	// of disk space.
+	fcopy, err := ioutil.TempFile("", "harbourbridge.data")
+	if err != nil {
+		return nil, 0, err
+	}
+	syscall.Unlink(fcopy.Name()) // File will be deleted when this process exits.
+	_, err = io.Copy(fcopy, f)
+	if err != nil {
+		return nil, 0, fmt.Errorf("can't write stdin to tmp file: %w", err)
+	}
+	_, err = fcopy.Seek(0, 0)
+	if err != nil {
+		return nil, 0, fmt.Errorf("can't reset file offset: %w", err)
+	}
+	n, err := getSize(fcopy)
+	return fcopy, n, nil
+}
+
+// createDatabase returns a newly create Spanner DB.
```

Comment:
Should be CreateDatabase to match case of function definition.

Reply:
updated Comment